packages/core/src/Machine.ts

```diff
@@ -26,26 +25,19 @@ export function Machine<
   config: MachineConfig<TContext, TStateSchema, TEvent>,
   options?: Partial<MachineOptions<TContext, TEvent>>,
   initialContext?: TContext
-): StateMachine<TContext, TStateSchema, TEvent>;
+): MachineNode<TContext, TStateSchema, TEvent>;
 export function Machine<
   TContext = DefaultContext,
   TStateSchema extends StateSchema = any,
   TEvent extends EventObject = AnyEventObject
 >(
   config: MachineConfig<TContext, TStateSchema, TEvent>,
-  options?: Partial<MachineOptions<TContext, TEvent>>,
-  initialContext: TContext | (() => TContext) | undefined = config.context
-): StateMachine<TContext, TStateSchema, TEvent> {
-  const resolvedInitialContext =
-    typeof initialContext === 'function'
-      ? (initialContext as (() => TContext))()
-      : initialContext;
-
-  return new StateNode<TContext, TStateSchema, TEvent, any>(
+  options?: Partial<MachineOptions<TContext, TEvent>>
+): MachineNode<TContext, TStateSchema, TEvent> {
+  return new MachineNode<TContext, TStateSchema, TEvent, any>(
     config,
-    options,
-    resolvedInitialContext
-  ) as StateMachine<TContext, TStateSchema, TEvent>;
+    options
+  ) as MachineNode<TContext, TStateSchema, TEvent>;
```

Comment:
this cast most likely is not needed anymore as you instantiate `MachineNode` which is the same as the expected return type, whereas previously instantiated class was different than the return type

That being said - not sure if using "node" in the name of this public type wont be slightly confusing, after all a user doesnt call into `MachineNode` or `createMachineNode`. Seems like the previous name (StateMachine) could have been more intuitive for users. Additionally - we can treat machine as a node, but at least in the SCXML it is not. Well - it's not described in SCXML in details, but when reading through the spec one can conclude it's not a node. It has different possible descendent elements and, what's even more important, one cannot transition to it. Thinking about nodes as the stuff we can transition to simplifies the mental model - states (nodes) represent possible configuration, they are directly represented on the visualized graph, whereas a machine is more like a wrapper containing the graph.

Reply:
Let's defer separating the root node to a different PR. Notice that the `<scxml>` children and `<state>` children have children in common:

- `<state>`
- `<parallel>`
- `<final>`

It might be redundant to model it differently than a compound `<state>` node, but let's explore this in a separate PR.